v2/azure-arm.yaml

```diff
@@ -1530,6 +1530,64 @@ objectModelConfiguration:
       AFDOriginProperties:
         HostName:
           $importConfigMapMode: optional
+  cognitiveservices:
+    2024-10-01:
+      Account:
+        $exportAs: Account
+        $supportedFrom: v2.15.0
+        $azureGeneratedSecrets:
+          - Key1
```

Comment:
These fields unfortunately don't magically get handled. This AzureGeneratedSecrets requires that you implement a hook/handler like [Azure Storage does](https://github.com/Azure/azure-service-operator/blob/main/v2/api/storage/customizations/storage_account_extensions.go#L34), where we call the API that serves the secrets and map those secrets into the underlying secret.

I'm not seeing where you've done that here.